x/staking/types/codec.go

```diff
@@ -15,6 +15,8 @@ func RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) {
 	cdc.RegisterConcrete(&MsgDelegate{}, "cosmos-sdk/MsgDelegate", nil)
 	cdc.RegisterConcrete(&MsgUndelegate{}, "cosmos-sdk/MsgUndelegate", nil)
 	cdc.RegisterConcrete(&MsgBeginRedelegate{}, "cosmos-sdk/MsgBeginRedelegate", nil)
+	// TODO: do we need this?
+	// cdc.RegisterConcrete(&Validator{}, "cosmos-sdk/Validator", nil)
```

Comment:
No, I don't think so.

Reply:
This is only if we want to support Amino, right?

Reply:
Without this line, amino will just marshal the Validator struct normally. E.g. in JSON, it'll look like:
```js
{
  "operator_address": "...",
  "consensus_pub_key": {"type": "cosmos/ed25519", "value": "{bytes}"},
  // other fields
}
```

If you add that line, I believe (not 100% sure, maybe an amino expert (!) can confirm), the JSON will look like:

```js
{
  "type": "cosmos-sdk/Validator", "value": "{bytes}"
}
```

Afaik we only we only RegisterConcrete for representing all the structs that implement a particular interface in Amino.